translate/convert/test_po2html.py

```diff
@@ -57,27 +57,57 @@ def test_linebreaks(self):
 </body>'''
         assert htmlexpected.replace("\n", " ") in self.converthtml(posource, htmlsource).replace("\n", " ")
 
-    @mark.xfail(reason="Not Implemented")
+    def test_replace_substrings(self):
+        """Should replace substrings correctly, issue #3416"""
+        htmlsource = '''<!DOCTYPE html><html><head><title>sub-strings substitution</title></head><body>
+<h2>This is heading 2</h2>
+<p>The heading says: <b>This is heading 2</b></p>
+</body></html>'''
+        posource = '#:html.body.h2:2-1\nmsgid "This is heading 2"\nmsgstr "Αυτή είναι μία Ετικέτα 2"\n\n#html.body.p:3-1\nmsgid "The heading says: <b>This is heading 2</b>"\nmsgstr "Η ετικέτα λέει: <b>Αυτή είναι μία Ετικέτα 2</b>"\n'
+        htmlexpected = '''<!DOCTYPE html><html><head><title>sub-strings substitution</title></head><body>
```

Comment:
So `title` elements are not yet translatable?

Reply:
yes they are; there should be a translation unit for the title as well.
however this test is only checking that we're not mixing up translations when one is a substring of the other.